resource.go

```diff
@@ -20,6 +22,7 @@ const (
 )
 
 type Config struct {
+	BackOffFactory      func() backoff.Interface
```

Comment:
Do we even need to configure that? Maybe an internal is a good fit here?

Reply:
It is configurable now. We default it. Good enough. Lets go.